src/signalrclient/connection_impl.cpp

```diff
@@ -56,16 +59,22 @@ namespace signalr
             // Signaling the event is safe here. We are in the dtor so noone is using this instance. There might be some
             // outstanding threads that hold on to the connection via a weak pointer but they won't be able to acquire
             // the instance since it is being destroyed. Note that the event may actually be in non-signaled state here.
-            m_start_completed_event.set();
-            shutdown().get();
-        }
-        catch (const pplx::task_canceled&)
-        {
-            // because we are in the dtor and the `connection_imp` is ref counted we should not get the `task_canceled`
-            // exception because it would indicate that some other thread/task still holds reference to this instance
-            // so how come we are in the dtor?
-            _ASSERTE(false);
-            return;
+            m_start_completed_event.cancel();
+            completion_event completion;
+            shutdown([completion](std::exception_ptr exception) mutable
+                {
+                    if (exception == nullptr)
+                    {
+                        completion.set();
+                    }
+                    else
+                    {
+                        // TODO: Log?
+                        completion.set(exception);
```

Comment:
Logging does seem better. This will cause the destructor to throw, right? I'm pretty sure that's a bad idea. http://blog.ycshao.com/2012/03/22/effective-c-item-8-prevent-exceptions-from-leaving-destructors/

Reply:
We explicitly catch everything in the destructor a couple lines down. Glad you're looking for that kind of thing though!

I guess we can remove setting the exception here so it won't even throw and just log.